### PR TITLE
WebUSB Support.

### PR DESCRIPTION
**Pull request by nodar-chkuaselidze**
WebUSB Support.

Notes:
 - webusb integration test requires user action to start, could not find flag that disables webusb security checks and lists all devices.

### CODE REVIEW

File: lib/devices/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/*!
+ * index.js - type definitions
+ * Copyright (c) 2019, Nodari Chkuaselidze (MIT License)
+ */
+
+'use strict';
+
+/**
+ * @module device
+ */
```
Comment (tynes) [inline lib/devices/index.js]: Were you planning on adding the typedefs here as part of this PR?

File: examples/webusb/index.js [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+    renderManager();
+  });
+
+  const pubkeyBtn = document.createElement('button');
+  pubkeyBtn.innerText = 'Get public key';
+  pubkeyBtn.addEventListener('click', async () => {
+    const device = manager.device;
+
+    if (!device) {
+      alert('WTF..');
```
Comment (tynes) [inline examples/webusb/index.js]: Maybe a better error message? haha

File: lib/devices/index.js (hunk 1 above)
Comment (nodar-chkuaselidze) [reply lib/devices/index.js]: I moved them in different branch that I have not published, I will update

File: examples/webusb/index.js (hunk 2 above)
Comment (nodar-chkuaselidze) [reply examples/webusb/index.js]: lol, forgot to change it. I will clean up comments :+1:

File: examples/webusb/index.js [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+/* eslint-env browser */
+
+const assert = require('bsert');
+const bledger = require('../..');
+const {LedgerBcoin} = bledger;
+const {Device} = bledger.WebUSB;
+const {DeviceInfo} = bledger.WebUSB;
+const KeyRing = require('bcoin/lib/primitives/keyring');
+
+const usb = navigator.usb;
```
Comment (tynes) [inline examples/webusb/index.js]: Maybe check to make sure that `navigator` exists first and returning early if it does not

File: lib/devices/webusb.js [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+ * @const {Number}
+ */
+const PACKET_SIZE = 64;
+
+/**
+ * USB instance
+ * @const {USBDevice}
+ */
+
+const usb = navigator.usb;
```
Comment (tynes) [inline lib/devices/webusb.js]: I think its a good idea to check for `navigator` here and stop the process early if it doesn't exist so people get a straight forward error message

File: .circleci/config.yml [hunk 5]
```diff
@@ -15,7 +15,7 @@ jobs:
           # fallback to using the latest cache if no exact match is found
           - v1-dependencies-
       - run: npm install
-      - run: npm install eslint nyc codecov
+      - run: npm install eslint babel-eslint nyc codecov
```
Comment (tynes) [inline .circleci/config.yml]: Should be able to install `bslint` instead of `babel-eslint`

**Pull request merged**